hal/source/bmp180_device.cpp

```diff
@@ -0,0 +1,336 @@
+#include "bmp180_device.h"
+
+Bmp180Device::Bmp180Device():
+	serial_(protocol_),
+	write_addr_(default_i2c_addr_),
+	read_addr_(default_i2c_addr_ | 0x01) {
+}
+
+Bmp180Device::~Bmp180Device() {
+
+}
+
+error_condition Bmp180Device::ReConfig(const CommonKeyPairs& config) {
+	ClearInitialized();
+
+	auto key_value = config.find(kPortName);
+	if (key_value == config.end())
+		return kInvalidArgument;
+	else {
+		if (serial_.ReConfig(config) != kNoError)
+			return kInvalidArgument;
+	}
+	SetInitialized();
+
+	if (GetSystemParam() != kNoError)
+		return kInvalidArgument;
+
+	if (GetCalibParam() != kNoError)
+		return kInvalidArgument;
+}
+
+error_condition Bmp180Device::DeConfig() {
+	return kNoError;
+}
+
+error_condition Bmp180Device::Set(const string& key, const unsigned int value) {
+	return kNoError;
+}
+
+error_condition Bmp180Device::Set(const string& key, \
+		const vector<uint8_t>& value) {
+	if (key != kI2CAddress)
+		return kInvalidArgument;
+
+	write_addr_ = value[0];
+	read_addr_ = value[0] | 0x01 ;
+
+	return kNoError;
+}
+
+error_condition Bmp180Device::Set(const string& key, const string& value) {
+	return kNoError;
+}
+
+error_condition Bmp180Device::Get(const string& key, unsigned int& result) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	if (key == kTemperature) {
+		int temperature;
+		if (GetTemperature(temperature) != kNoError) {
+			return make_error_condition(errc::io_error);
+		}
+		result = static_cast<unsigned int>(temperature);
+	} else if (key == kPressure) {
+		int pressure;
+		if (GetPressure(pressure) != kNoError) {
+			return make_error_condition(errc::io_error);
+		}
+		result = static_cast<unsigned int>(pressure);
+	}
+	return kNoError;
+}
+
+error_condition Bmp180Device::Get(const string& key, vector<uint8_t>& result) {
+	return kNoError;
+}
+
+error_condition Bmp180Device::Get(const string& key, string& value) {
+	return kNoError;
+}
+
+error_condition Bmp180Device::BusRead(uint8_t reg, uint8_t len, \
+		vector<uint8_t>& read_data) {
+	vector<uint8_t> buffer{read_addr_, reg, len};
+	if (serial_.Set(SerialPort::kData, buffer) != kNoError)
+		return make_error_condition(errc::io_error);
+
+	DelayMSec(5);
+
+	if (serial_.Get(SerialPort::kData, read_data) != kNoError)
+		return make_error_condition(errc::io_error);
+
+	return kNoError;
+}
+
+error_condition Bmp180Device::BusWrite(uint8_t reg, \
+		vector<uint8_t>& write_data) {
+	vector<uint8_t> buffer{write_addr_, reg};
```

Comment:
Does this pass the syntax check? I'm thinking if it should be:
```suggestion
	vector<uint8_t> buffer = {write_addr_, reg};
```